### DISCUSSION TITLE
Up and down casting of smart pointers

### DISCUSSION THREAD

**Discussion by domcir** [bytedeco/javacpp#686] [Q&A]
Hi

When I use a facroy to get a smart pointer object how should I handle up and down casting of that smart pointer in Java?

Something like 'std::static_pointer_cast' in C++.

Do I need to create a converter function that would look like 'std::smart_ptr<Base> convert(std::smart_ptr<Derived>)' or is there a better way?

Thanks
Dom

**Comment by saudet** [comment]
That's the essence of it, but we can deal with those more easily with the new `Info.upcast()` that @HGuillemet has contributed, see pull https://github.com/bytedeco/javacpp/pull/671.
